tests/cli/ursula/test_blockchain_ursula.py

```diff
@@ -0,0 +1,333 @@
+        miner.confirm_activity()
+
+    expiration = maya.now() + datetime.timedelta(days=3)
+    blockchain_policy = alice.grant(bob=bob,
+                                    label=random_policy_label,
+                                    m=1, n=1,
+                                    value=POLICY_VALUE,
+                                    expiration=expiration,
+                                    handpicked_ursulas={staking_participant})
+
+    # Bob joins the policy
+    bob.join_policy(random_policy_label, bytes(alice.stamp))
+
+    # Enrico Encrypts (of course)
+    enrico = Enrico(policy_encrypting_key=blockchain_policy.public_key,
+                    network_middleware=MockRestMiddleware())
+
+    for index, _period in enumerate(range(half_stake_time-5)):
+        logger.debug(f"period {current_period}")
+
+        alphabet = string.ascii_letters + string.digits
+
+        # Random Request Periods
+        if not random.choice((True, False)):
+            continue  # maybe re-encrypt
+
+        max_reencryptions_per_period = 5
+        quantity = random.choice(range(max_reencryptions_per_period+1))
+        quantity *= index  # factorial or 0
+        verifying_key = UmbralPublicKey.from_bytes(bytes(alice.stamp))
+
+        # Random Re-encryptions
+        for _i in range(quantity):
+
+            # Encrypt
+            random_data = ''.join(secrets.choice(alphabet) for i in range(secrets.choice(range(20, 100))))
+            ciphertext, signature = enrico.encrypt_message(message=bytes(random_data, encoding='utf-8'))
+
+            # Retrieve
+            payload = dict(message_kit=ciphertext,
+                           data_source=enrico,
+                           alice_verifying_key=verifying_key,
+                           label=random_policy_label)
+
+            _cleartext = bob.retrieve(**payload)
+
+        # Ursula Staying online and the clock advancing
+        blockchain.time_travel(periods=1)
+        miner.confirm_activity()
+        current_period += 1
+
+    # Finish the passage of time... once and for all
+    for _ in range(5):
+        current_period += 1
+        logger.debug(f"period {current_period}")
+        blockchain.time_travel(periods=1)
+        miner.confirm_activity()
+
+    #
+    # WHERES THE MONEY URSULA?? - Collecting Rewards
+    #
+
+    # Snag a random teacher from the fleet
+    random_teacher = list(blockchain_ursulas).pop()
+
+    collection_args = ('--mock-networking',
+                       'ursula', 'collect-reward',
+                       '--teacher-uri', random_teacher.rest_interface,
+                       '--config-file', configuration_file_location,
+                       '--poa',
+                       '--force')
+
+    result = click_runner.invoke(nucypher_cli,
+                                 collection_args,
+                                 input=INSECURE_DEVELOPMENT_PASSWORD,
+                                 catch_exceptions=False)
+    assert result.exit_code == 0
```

Comment:
Where did the tests whether Ursula collected the correct amount go?

Reply:
This Issue is somewhat blocked by the need for gas control for tests and in production.

Reply:
It definitely feels missing.  I'll look into what's needed. I can use your help on this one @szotov.

Reply:
If it's a quick fix, the tests for quantity can be included here.
If not, you can include a placeholder comment, open an issue and merge this PR without it

Reply:
:heavy_check_mark: